core/src/storage/persistent_state/shard_state/writer.rs

```diff
@@ -29,6 +30,14 @@ impl<'a> ShardStateWriter<'a> {
     // Partially written BOC file.
     const FILE_EXTENSION_TEMP: &'static str = "boc.temp";
 
+    pub fn file_name_str(name: &str) -> PathBuf {
+        PathBuf::from(name).with_extension(Self::FILE_EXTENSION)
+    }
+
+    pub fn temp_file_name_str(name: &str) -> PathBuf {
+        PathBuf::from(name).with_extension(Self::FILE_EXTENSION_TEMP)
+    }
+
     pub fn file_name(block_id: &BlockId) -> PathBuf {
         PathBuf::from(block_id.to_string()).with_extension(Self::FILE_EXTENSION)
     }
```

Comment:
will work for both block_id and strings and whatever impls display

```suggestion
  pub fn file_name<S>(block_id: S) -> PathBuf
    where
        S: Display,
    {
        PathBuf::from(block_id.to_string()).with_extension(Self::FILE_EXTENSION)
    }
```